src/Processor/ProductOptionValue/ReferenceEntityOptionValuesProcessor.php

```diff
@@ -0,0 +1,175 @@
+use Sylius\Component\Product\Model\ProductOptionValueTranslationInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Synolia\SyliusAkeneoPlugin\Component\Attribute\AttributeType\ReferenceEntityAttributeType;
+use Synolia\SyliusAkeneoPlugin\Entity\ApiConfiguration;
+use Synolia\SyliusAkeneoPlugin\Exceptions\ApiNotConfiguredException;
+use Synolia\SyliusAkeneoPlugin\Manager\ProductOptionManager;
+use Synolia\SyliusAkeneoPlugin\Provider\AkeneoAttributePropertiesProvider;
+use Synolia\SyliusAkeneoPlugin\Task\AttributeOption\CreateUpdateDeleteTask;
+use Synolia\SyliusAkeneoPlugin\Transformer\ProductOptionValueDataTransformerInterface;
+use Webmozart\Assert\Assert;
+
+class ReferenceEntityOptionValuesProcessor extends AbstractOptionValuesProcessor
+{
+    /** @var \Akeneo\PimEnterprise\ApiClient\AkeneoPimEnterpriseClientInterface */
+    private $client;
+
+    /** @var \Synolia\SyliusAkeneoPlugin\Provider\AkeneoAttributePropertiesProvider */
+    private $akeneoAttributePropertiesProvider;
+
+    /** @var \Sylius\Component\Resource\Repository\RepositoryInterface */
+    private $apiConfigurationRepository;
+
+    /** @var \Sylius\Component\Resource\Repository\RepositoryInterface */
+    private $localeRepository;
+
+    public function __construct(
+        RepositoryInterface $productOptionValueRepository,
+        RepositoryInterface $productOptionValueTranslationRepository,
+        FactoryInterface $productOptionValueFactory,
+        FactoryInterface $productOptionValueTranslationFactory,
+        LoggerInterface $akeneoLogger,
+        EntityManagerInterface $entityManager,
+        AkeneoPimEnterpriseClientInterface $client,
+        AkeneoAttributePropertiesProvider $akeneoAttributePropertiesProvider,
+        RepositoryInterface $apiConfigurationRepository,
+        ProductOptionValueDataTransformerInterface $productOptionValueDataTransformer,
+        RepositoryInterface $localeRepository
+    ) {
+        parent::__construct(
+            $productOptionValueRepository,
+            $productOptionValueTranslationRepository,
+            $productOptionValueFactory,
+            $productOptionValueTranslationFactory,
+            $akeneoLogger,
+            $entityManager,
+            $productOptionValueDataTransformer
+        );
+
+        $this->client = $client;
+        $this->akeneoAttributePropertiesProvider = $akeneoAttributePropertiesProvider;
+        $this->apiConfigurationRepository = $apiConfigurationRepository;
+        $this->localeRepository = $localeRepository;
+    }
+
+    public static function getDefaultPriority(): int
+    {
+        return 90;
+    }
+
+    public function support(AttributeInterface $attribute, ProductOptionInterface $productOption, array $context = []): bool
+    {
+        return ReferenceEntityAttributeType::TYPE === $attribute->getType() && $this->isEnterprise();
+    }
+
+    private function isEnterprise(): bool
```

Comment:
IMO, this function shoundn't be here.

Reply:
fixed in commit 3f521e9319e96e2aca784b1d4b43965bccca3fb5